### ISSUE TITLE
`tf.keras.layers.RNN` is very different from `tf.nn.dynamic_rnn`

### ISSUE THREAD

**Issue by siyuany** [tensorflow/tensorflow#30512]
In the documentation of [`tf.nn.dynamic_rnn`](https://www.tensorflow.org/api_docs/python/tf/nn/dynamic_rnn), it gives such a warning saying:

> Warning: THIS FUNCTION IS DEPRECATED. It will be removed in a future version. 
> Instructions for updating: Please use keras.layers.RNN(cell), which is equivalent to this API

BUT when I checked [`tf.keras.layers.RNN(cell)`](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RNN), I found an important feature of `tf.nn.dynamic_rnn` is lost, which enables users to deal with sequences of variable lengths.

```
tf.nn.dynamic_rnn(
    cell,
    inputs,
    sequence_length=None,
    initial_state=None,
    dtype=None,
    parallel_iterations=None,
    swap_memory=False,
    time_major=False,
    scope=None
)
```

Here `sequence_length` can be a tensor which indicating the length of the sequences. Let's see the API of `tf.keras.layers.RNN`:

```
__init__(
    cell,
    return_sequences=False,
    return_state=False,
    go_backwards=False,
    stateful=False,
    unroll=False,
    time_major=False,
    **kwargs
)
```

There is actually no such an argument. If I want to deal with variant-length sequences, I have to do a lot of work myself. So I still would like to use `tf.nn.dynamic_rnn` rather than `tf.keras.layers.RNN`.

After all, I think as the arguments and funcationality is different, 

> keras.layers.RNN(cell), which is equivalent to this API

is really a very big mistake. And please keep `tf.nn.dynamic_rnn` and remove it from deprecated functions.

**Comment by zakizhou** [comment]
There is an arg called `mask` in the `__call__` of `tf.keras.layers.RNN`, you can build a mask from `sequence_length `(use `tf.sequence_mask` to do so) and pass the mask to this arg. BTW, subclasses of `tf.keras.layers.RNN` like `tf.keras.layers.GRU` can work with GPU and cuda, but `tf.nn.dynamic_rnn` only works with CPU.

**Comment by yxd886** [comment]
how about the feature swap_memory? it seems that tf.keras.layers.RNN does not provide this flag
